### DISCUSSION TITLE
Integration Issue with SIWE and Auth.js Credentials Provider

### DISCUSSION THREAD

**Discussion by neotheprogramist** [nextauthjs/next-auth#8916] [Ideas]
### Provider type

Credentials

### Environment

  System:
    OS: Linux 6.4 Fedora Linux 38 (Workstation Edition)
    CPU: (16) x64 AMD Ryzen 7 PRO 6850U with Radeon Graphics
    Memory: 6.39 GB / 14.38 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 18.17.1 - /usr/bin/node
    npm: 9.8.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 117.0.5938.92
  Packages:
    "@auth/core": "^0.13.0",
    "@builder.io/qwik": "^1.2.12",
    "@builder.io/qwik-auth": "0.1.3",
    "@builder.io/qwik-city": "^1.2.12",

### Reproduction URL

https://github.com/visoftsolutions/ksox-web-old/tree/authjs-credential-dead-end/exchange

### Describe the issue

I'm facing a problem when trying to integrate Sign-In with Ethereum (SIWE) with Auth.js as per this [guide](https://blog.spruceid.com/sign-in-with-ethereum-on-next-js-applications/). There's no straightforward way to pass the 'message' and 'signature' credentials from SIWE to Auth.js's Credential provider programmatically. The current implementation redirects users to a page where they're prompted to enter these credentials manually.

### How to reproduce

In the repo https://github.com/visoftsolutions/ksox-web-old/tree/authjs-credential-dead-end

run redis db
```shell
podman run -p 6379:6379 -p 8001:8001 -it redis/redis-stack:latest
```

run dev
```shell
npm run dev -w exchange
```

and go to http://localhost:5173/siwe/

### Expected behavior

I'd like to pass the SIWE credentials to Auth.js's Credential provider programmatically, without redirecting users for manual input.
Solving this would make Qwik a more viable option for web3 applications alongside React.

**Comment by ThangHuuVu** [comment]
your repro link is a 404
I'm curious to hear about how do you think we can improve the core to support this use-case.

**Reply by neotheprogramist** [reply]
Fixed that.
I think we need some way to pass credentials programmatically to Credential Provider.
Similar how NextAuth does that [here](https://docs.login.xyz/integrations/nextauth.js).
We need something like "redirect: false" to make that work in my opinion.
```
signIn("credentials", {
  message: JSON.stringify(message),
  redirect: false,
  signature,
  callbackUrl,
})
```
I need to follow this tutorial but with [Qwik Framework](https://qwik.builder.io/) instead of Next
